include/rmm/resource_ref.hpp

```diff
@@ -65,5 +65,21 @@ using host_device_resource_ref =
 using host_device_async_resource_ref =
   cuda::mr::async_resource_ref<cuda::mr::host_accessible, cuda::mr::device_accessible>;
 
+/**
+ * @brief Convert pointer to memory resource into `device_async_resource_ref`, checking for
+ * `nullptr`
+ *
+ * @tparam Resource The type of the memory resource.
+ * @param res A pointer to the memory resource.
+ * @return A `device_async_resource_ref` to the memory resource.
+ * @throws std::logic_error if the memory resource pointer is null.
+ */
+template <class Resource>
+device_async_resource_ref to_device_async_resource_ref_checked(Resource* res)
+{
+  RMM_EXPECTS(nullptr != res, "Unexpected null resource pointer.");
```

Comment:
This comes from the old advice to use `if (CONSTANT == variable)` rather than `if (variable == CONSTANT)`. The reasoning was that it was common to typo `=` for `==`, leading to hard to find assignment bugs. These days we have tools to detect that, but old habits die hard.

If I follow ancient practices, maybe you *should* call me Yoda. :)

Reply:
Ha! This is non-blocking, of course. You are the wisest Jedi so I defer to your preference.

Reply:
In this case went with the real dinosaur code since the error string makes it obvious. :)